LambdaEngine/Include/ECS/ComponentStorage.h

```diff
@@ -25,32 +25,32 @@ namespace LambdaEngine
 		template<typename Comp>
 		Comp& GetComponent(Entity entity);
 
-		bool DeleteComponent(Entity entity, std::type_index componentType);
+		bool DeleteComponent(Entity entity, const ComponentType* pComponentType);
 
 		template<typename Comp>
 		bool HasType() const;
 
-		bool HasType(std::type_index componentType) const;
+		bool HasType(const ComponentType* pComponentType) const;
 
-		IComponentArray* GetComponentArray(std::type_index componentType);
-		const IComponentArray* GetComponentArray(std::type_index componentType) const;
+		IComponentArray* GetComponentArray(const ComponentType* pComponentType);
+		const IComponentArray* GetComponentArray(const ComponentType* pComponentType) const;
 
 		template<typename Comp>
 		ComponentArray<Comp>* GetComponentArray();
 
 	private:
-		std::unordered_map<std::type_index, uint32> m_CompTypeToArrayMap;
+		std::unordered_map<const ComponentType*, uint32> m_CompTypeToArrayMap;
```

Comment:
Is `ComponentType::operator()` or the pointer being used for mapping?

Reply:
Yes the `ComponentType::operator()` is the hasher used by default